Framework/Core/test/test_DataAllocator.cxx

```diff
@@ -145,6 +145,13 @@ DataProcessorSpec getSourceSpec()
     ASSERT_ERROR(messageablevector.size() == 0);
     messageablevector.push_back(a);
     messageablevector.emplace_back(10, 20, 0xacdc);
+
+    // make a PMR std::vector, make it large to test the auto transport buffer resize funtionality as well
+    Output pmrOutputSpec{"TST", "PMRTESTVECTOR", 0};
```

Comment:
Thank you for the test.

Do I understand correctly you could have done:

```C++
auto pmrvector = pc.outputs().makeVector<o2::test::TriviallyCopyable>(pmrOutputSpec);
pmrvec.emplace_back(o2::test::TriviallyCopyable{1, 2, 3});
```

to obtain the same result? This well illustrates my point: what is the advantage of `getMemoryResource` + `adoptContainer` vs `makeVector`?

Reply:
I could have used makeVector, but a call to adoptContainer is needed anyway to make the DPL aware of the data (i guess it is equivalent to fairmq Send() or so). makeVector is just a shorthand for `auto pmrvec = o2::vector<o2::test::TriviallyCopyable>(pc.outputs().getMemoryResource(pmrOutputSpec));` - I'm actually not sure makeVector is needed since it only gives you a std::vector... all the other code is generic and will work with other (compliant) containers.
I have not put it in you make<>  since in the end the semantics are different: yours returns a ref to a managed resource, mine returns the container itself (a handle) which is scope managed until Send() or adoptContainer() is called. I'll add a test for makeVector() as well to have better coverage.